wdl/transforms/new-base/src/main/scala/wdl/transforms/base/linking/expression/values/EngineFunctionEvaluators.scala

```diff
@@ -595,17 +595,19 @@ object EngineFunctionEvaluators {
                                ioFunctionSet: IoFunctionSet,
                                forCommandInstantiationOptions: Option[ForCommandInstantiationOptions])
                               (implicit expressionValueEvaluator: ValueEvaluator[ExpressionElement]): ErrorOr[EvaluatedValue[WomString]] = {
-      def simpleBasename(fileNameAsString: WomString) = fileNameAsString.valueString.split('/').last
+      def simpleBasename(fileNameAsString: WomString): Try[String] = {
+        Try(Await.result(ioFunctionSet.resolvedFileBasename(fileNameAsString.valueString), 60.seconds))
```

Comment:
*cough* blocking in production code *cough* 😛

Reply:
This is almost certainly a case of "but everyone else is doing it!"... I'm copying from other functions here which seem to be grandfathering-in "we didn't even know about Futures yet" synchronization 😬

Reply:
Hey at least we knew about `Try` back then!

Reply:
At least it has a non-infinite timeout!

Reply:
`Duration.Infinite - 1`